### PR TITLE
[com_banners] use regexp()

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #26788 .

### Summary of Changes
used the  query method  `regexp()`

### Testing Instructions
Enter some meta keywords in Global Configuration.
Create a banner, enter same meta keywords.
Publish a banner module and enable filtering by meta keywords.

### Expected result
Banner not shown.

### Actual result

SQL error

### CODE REVIEW

**Comment by SharkyKZ** [comment]
Doesn't work:

>42601, 7, ERROR: syntax error at or near "[" LINE 4: ...0 AND cl.own_prefix=0 AND 1) AND (a.metakey ~* [[:<:]]Joo... ^

**Comment by alikon** [comment]
i'm afraid i've missed to  quote `[[:<:]]`   ....  was in hurry.....

**Comment by SharkyKZ** [comment]
No difference.

>42601, 7, ERROR: syntax error at or near "Joomla" LINE 4: ....own_prefix=0 AND 1) AND (a.metakey ~* '[[:<:]]'Joomla'[[:... ^

File: components/com_banners/models/banners.php [hunk 1]
```diff
@@ -161,16 +161,17 @@ protected function getListQuery()
 						. ' AND cl.own_prefix=0 '
 						. ' AND ' . ($prefix == substr($keyword, 0, strlen($prefix)) ? '1' : '0');
 
-					$condition2 = "a.metakey REGEXP '[[:<:]]" . $db->escape($keyword) . "[[:>:]]'";
+					$regexp = $db->quote("[[:<:]]") . $db->escape($keyword) . $db->quote("[[:>:]]");
```
Comment (SharkyKZ) [inline components/com_banners/models/banners.php]: This should work.
```suggestion
					$regexp = $db->quote("[[:<:]]" . $db->escape($keyword) . "[[:>:]]");
```

Comment (alikon) [reply components/com_banners/models/banners.php]: yes just haved  some time to run an sql fiddle  😃

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on ed41cd811633b622ac371cc4d401eef2f2e78894<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26791">issues.joomla.org/tracker/joomla-cms/26791</a>.</sub>

**Comment by waader** [comment]
I tested with postgresql and the nightly build of j39. Following your test instructions I get the following errors:

Warning: Invalid argument supplied for foreach() in joomla39\components\com_banners\models\banners.php on line 208

Warning: Invalid argument supplied for foreach() in joomla39\modules\mod_banners\tmpl\default.php on line 19

**Comment by alikon** [comment]
closing as the same fix submitted to  #26875

**Pull request closed without merging**